internal/models/validator.go

```diff
@@ -0,0 +1,23 @@
+package models
+
+import (
+	"context"
+
+	"github.com/go-playground/validator/v10"
+	"golang.org/x/crypto/ssh"
+)
+
+var validate *validator.Validate
+
+func init() {
+	validate = validator.New()
+
+	_ = validate.RegisterValidation("sshPubkey", func(fl validator.FieldLevel) bool {
+		_, err := ssh.ParsePublicKey(fl.Field().Bytes())
+		return err == nil
+	})
+}
+
+func Validate(ctx context.Context, model interface{}) error {
+	return validate.StructCtx(ctx, model)
```

Comment:
Should we use errors.new or create a type implementing Error interface?

Reply:
the validate.StructCtx alreay returns type that impements error interface, with some additional goodies, but it already is an error, we could state clearly what we return here tho, as we know it won't be a generic error